plot_krun_results.py

```diff
@@ -80,65 +88,88 @@
 DPI = 300
 
 
-def main(is_interactive, data_dcts, window_size, outfile, xlimits, single_exec,
-         with_outliers, unique_outliers, mean=False, sigma=False,
-         benchmark=None):
-    """Create a new dictionary in bmark->machine->data format.
+def main(is_interactive, data_dcts, plot_titles, window_size, outfile,
+         xlimits, with_outliers, unique_outliers, mean=False, sigma=False,
+         one_page=False):
+    """Determine which plots to put on each page of output.
     Plot all data.
     """
-    pdf = None  # PDF output (non-interactive mode).
+
+    pdf = None  # PDF output (for non-interactive mode).
 
     if not is_interactive:
         pdf = PdfPages(outfile)
         set_pdf_metadata(pdf)
 
-    benchmark_in_results = False
-    for machine in data_dcts:
-        keys = sorted(data_dcts[machine]['data'].keys())
-        if benchmark in keys:
-            benchmark_in_results = True
-        for key in keys:
-            bench, vm, variant = key.split(':')
-            executions = data_dcts[machine]['data'][key]
-            if benchmark is None and len(executions) == 0:
-                print('Skipping: %s (no executions)' % key)
-            elif benchmark is None and len(executions[0]) == 0:
-                print('Skipping: %s (benchmark crashed)' % key)
-            elif (benchmark is not None) and key != benchmark:
-                continue
-            else:
+    # Run sequences, outliers and subplot titles for each page we need to plot.
+    pages, all_subplot_titles = list(), list()
+    all_outliers, all_common, all_unique = list(), list(), list()
+
+    # By default, each benchmark from each machine is placed on a separate
+    # page. If the --one-page switch has been passed in, then we place
+    # all plots on a single page. In which case, we need to construct
+    # a flat list of all run sequences.
+    if one_page:
+        page, subplot_titles = list(), list()
+        page_common, page_unique, page_outlier = list(), list(), list()
+        for machine in data_dcts['data']:
+            for key in data_dcts['data'][machine]:
+                for index, run_seq in enumerate(data_dcts['data'][machine][key]):
+                    page.append(run_seq)
+                    subplot_titles.append(plot_titles[machine][key][index])
+                    if with_outliers:
+                        page_outlier.append(data_dcts['outliers'][machine][key][index])
+                    else:
+                        page_outlier.append(None)
+                    if unique_outliers:
+                        page_common.append(data_dcts['common_outliers'][machine][key][index])
+                        page_unique.append(data_dcts['unique_outliers'][machine][key][index])
+                    else:
+                        page_common.append(None)
```

Comment:
Not saying this is wrong, but at a glance it's not clear why we add `Nones` into the lists. Maybe scatter a few more comments? Thoughts?